text/0000-wotsicide/0000-wotsicide.md

```diff
@@ -0,0 +1,387 @@
+        <td>The hash of the milestone which created the referenced <i>TreasuryOutput</i>.</td>
+    </tr>
+</table>
+
+### Treasury Output
+
+The `TreasuryOutput` is a special output type which represents the treasury of the network, respectively the non yet migrated funds. At any given moment in time, there is only one `TreasuryOutput`.
+
+Serialized form:
+
+<table>
+    <tr>
+        <td><b>Name<b></td>
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Output Type</td>
+        <td>byte</td>
+        <td>
+            Set to <strong>value 2</strong> to denote an <i>TreasuryInput</i>.
+        </td>
+    </tr>
+    <tr>
+        <td>Amount</td>
+        <td>uint64</td>
+        <td>The amount of funds residing in the treasury.</td>
+    </tr>
+</table>
+
+The `TreasuryOutput` can not be referenced or spent by transactions, it can only be referenced by receipts.
+
+The `TreasuryOutput` can be queried from the HTTP API and needs to be included within snapshots in order to keep the total supply intact.
+
+## Receipts
+
+Receipts allow for fast migration of funds from the legacy into the new network by representing entries of funds which were migrated in the old network.
+
+### Schema
+Receipts are listings of funds for which nodes must generate UTXOs in the form of `SigLockedSingleOutput`s targeting the given address. Receipts are embedded within milestone payloads and therefore signed by the Coordinator. A milestone may contain up to one receipt as a payload. The Coordinator chooses whether to embed a receipt payload or not.
+
+Serialized form:
+
+<table>
+    <tr>
+        <td><b>Name<b></td>
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Payload Type</td>
+        <td>uint32</td>
+        <td>Set to <strong>value 3</strong> to denote a <i>Receipt</i>.</td>
+    </tr>
+    <tr>
+        <td>Migrated At</td>
+        <td>uint32</td>
+        <td>The index of the legacy milestone in which the listed funds were migrated at.</td>
+    </tr>
+    <tr>
+        <td>Final</td>
+        <td>bool</td>
+        <td>Flags whether this receipt is the last receipt for the given <i>Migrated At</i> index.</td>
+    </tr>
+    <tr>
+        <td>Funds Count</td>
+        <td>uint16</td>
+        <td>Denotes how many migrated fund entries are within the receipt.</td>
+    </tr>
+    <tr>
+        <td>Funds</td>
+        <td colspan="2">
+        <details open="true">
+            <summary>Migrated Funds Entry</summary>
+            <table>
+                <tr>
+                    <th>Name</th>
+                    <th>Type</th>
+                    <th>Description</th>
+                </tr>
+                <tr>
+                    <td>Tail Transaction Hash</td>
+                    <td>Array<byte>[49]</td>
+                    <td>
+                    The tail transaction hash of the bundle in which these funds were migrated.
```

Comment:
Even if it is not needed at all, wouldn't it still be a good idea to include a list of addresses of the old network, that were the input for the migration bundle. This way it would be easier to proof where these funds originally come from (e.g. for tax reasons etc). :shrug: They users would only need to query a chrysalis node to get this information.

Reply:
They can use a legacy node to query this information, there isn't really a need to bloat receipts with such data which isn't necessary for the actual migration.